internal/storage/bucket_handle_test.go

```diff
@@ -467,3 +470,318 @@ func (t *BucketHandleTest) TestUpdateObjectMethodWithMissingObject() {
+			ContentType: ContentType,
+			Metadata: map[string]string{
+				MetaDataKey: MetaDataValue,
+			},
+			ContentLanguage:    ContentLanguage,
+			ContentEncoding:    ContentEncoding,
+			CacheControl:       CacheControl,
+			ContentDisposition: ContentDisposition,
+			CustomTime:         CustomTime,
+			EventBasedHold:     true,
+			StorageClass:       StorageClass,
+			Acl:                nil,
+		})
+
+	AssertEq(nil, err)
+	// Validation of srcObject to ensure that it is not effected.
+	rc, err = t.bucketHandle.NewReader(context.Background(),
+		&gcs.ReadObjectRequest{
+			Name: TestSubObjectName,
+			Range: &gcs.ByteRange{
+				Start: uint64(0),
+				Limit: uint64(len(ContentInTestSubObject)),
+			},
+		})
+	AssertEq(nil, err)
+	defer rc.Close()
+	srcObjBuf := make([]byte, len(ContentInTestSubObject))
+	_, err = rc.Read(srcObjBuf)
+	AssertEq(nil, err)
+	// Reading content of destination object
+	rc, err = t.bucketHandle.NewReader(context.Background(),
+		&gcs.ReadObjectRequest{
+			Name: TestObjectName,
+			Range: &gcs.ByteRange{
+				Start: uint64(0),
+				Limit: uint64(composedObj.Size),
+			},
+		})
+	AssertEq(nil, err)
+	defer rc.Close()
+	dstObjBuf = make([]byte, composedObj.Size)
+	_, err = rc.Read(dstObjBuf)
+	AssertEq(nil, err)
+	// Destination object's content will get overwrite by srcObject.
+	ExpectEq(string(srcObjBuf[:]), string(dstObjBuf[:]))
+	AssertNe(nil, composedObj)
+	AssertEq(srcObj.Size, composedObj.Size)
+}
+
+func (t *BucketHandleTest) TestComposeObjectMethodWithOneSrcObject() {
+	var notfound *gcs.NotFoundError
+	// Checking that dstObject does not exist
+	_, err := t.bucketHandle.StatObject(context.Background(),
+		&gcs.StatObjectRequest{
+			Name: dstObjectName,
+		})
+	AssertTrue(errors.As(err, &notfound))
+	srcObj, err := t.bucketHandle.StatObject(context.Background(),
+		&gcs.StatObjectRequest{
+			Name: TestObjectName,
+		})
+	AssertEq(nil, err)
+	AssertNe(nil, srcObj)
+
+	composedObj, err := t.bucketHandle.ComposeObjects(context.Background(),
+		&gcs.ComposeObjectsRequest{
+			DstName:                       dstObjectName,
+			DstGenerationPrecondition:     nil,
+			DstMetaGenerationPrecondition: nil,
+			Sources: []gcs.ComposeSource{
+				{
+					Name: TestObjectName,
+				},
+			},
+			ContentType: ContentType,
+			Metadata: map[string]string{
+				MetaDataKey: MetaDataValue,
+			},
+			ContentLanguage:    ContentLanguage,
+			ContentEncoding:    ContentEncoding,
+			CacheControl:       CacheControl,
+			ContentDisposition: ContentDisposition,
+			CustomTime:         CustomTime,
+			EventBasedHold:     true,
+			StorageClass:       StorageClass,
+			Acl:                nil,
+		})
+
+	AssertEq(nil, err)
+	// Validation of srcObject to ensure that it is not effected.
+	rc, err := t.bucketHandle.NewReader(context.Background(),
+		&gcs.ReadObjectRequest{
+			Name: TestObjectName,
+			Range: &gcs.ByteRange{
+				Start: uint64(0),
+				Limit: uint64(len(ContentInTestSubObject)),
+			},
+		})
+	AssertEq(nil, err)
```

Comment:
can you move all this validation logic into private method, so it can be reused by all test method.

Reply:
fixed